src/controller/artifact/annotation/parser.go

```diff
@@ -0,0 +1,76 @@
+package annotation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/goharbor/harbor/src/lib/log"
+
+	"github.com/goharbor/harbor/src/pkg/artifact"
+)
+
+const (
+	// XIcon is icon content type image/x-icon
+	XIcon = "image/x-icon"
+	// GIF is icon content type image/gif
+	GIF = "image/gif"
+	// PNG is icon content type image/png
+	PNG = "image/png"
+	// JPEG is icon content type image/jpeg
+	JPEG = "image/jpeg"
+
+	// AnnotationPrefix is the prefix of annotation
+	AnnotationPrefix = "io.goharbor.artifact"
+
+	// SkipList is the key word of skip-list annotation
+	SkipList = "skip-list"
+	// Icon is the key word of icon annotation
+	Icon = "icon"
+)
+
+var (
+	// Registry for registered annotation parsers
+	Registry = map[string]Parser{}
+
+	// SortedAnnotationVersionList define the order of AnnotationParser from low to high version.
+	// Low version annotation parser will parser annotation first.
+	SortedAnnotationVersionList = make([]string, 0)
+)
+
+// NewParser creates a new annotation parser
+func NewParser() Parser {
+	return &parser{}
+}
+
+// Parser parses annotations in artifact manifest
+type Parser interface {
+	// Parse parses annotations in artifact manifest, abstracts data from artifact config layer into the artifact model
+	Parse(ctx context.Context, artifact *artifact.Artifact, manifest []byte) (err error)
+}
+
+type parser struct{}
+
+func (p *parser) Parse(ctx context.Context, artifact *artifact.Artifact, manifest []byte) (err error) {
```

Comment:
manifest will be the same one even there are multiple parsers are registered. Why not use an object ref as an argument and decode the bytes here only once?

Reply:
In the future may be we will support index. So here use []byte for extension. /cc @ywk253100

Reply:
ok